CloudAEye Integration with Datadog (ECOINT-19) (#2386)
DataDog/integrations-extras: cloudaeye/README.md

# CloudAEye

## Overview

> Kosal automates root cause analysis (RCA), significantly improving incident resolution time and streamlining the experience for engineers. This AI-powered workflow replaces manual tasks that engineers currently handle by analyzing data - logs, metrics, traces - and pin-pointing the root cause with relevant context. As a result, engineers can redirect their focus to more critical aspects. Additionally, Kosal adeptly identifies historical issues within Jira and traces related code changes that may lie at the heart of the problem.

Reviewer: 3a. what data are you sending in

> Kosal automates root cause analysis (RCA), significantly improving incident resolution time and streamlining the experience for engineers. This AI-powered workflow replaces manual tasks that engineers currently handle by analyzing data - logs, metrics, traces - and pin-pointing the root cause with relevant context. As a result, engineers can redirect their focus to more critical aspects. Additionally, Kosal adeptly identifies historical issues within Jira and traces related code changes that may lie at the heart of the problem.

Reviewer: This should be very clear and clear enough for a new customer understands what you product does, wht data is sent to datadog, and what you are pulling out (if applicable)

## Setup

### Installation

> Visit [CloudAEye][4] to sign up for free. Once registered, visit the [CloudAEye integrations page][1] and add a Datadog integration. This guides you through the Datadog OAUTH2 flow to grant CloudAEye access to your traces, logs, and metrics data.

Reviewer: OAuth needs to be intiated within Datadog. Please update the steps users will take to set up the integration

# CloudAEye

## Overview

> Kosal from CloudAEye is your root cause co-pilot. It analyzes your observability data (logs, metrics, traces) and pin-points the root cause automatically. Kosal publishes events with a summary to an integrated dashboard that helps to identify incidents faster. Additionally, Kosal adeptly identifies similar issues within Jira and related code changes that may be causing the incident. Using this integration you may troubleshoot and resolve incidents faster which improves your mean time to repair (MTTR).

Reviewer: Are the events a summary of the root cause or of the incident?

## Setup

### Installation

- Search for CloudAEye in the integrations page.
- Click on **Install Integration** button.
- Once installed, click on **Connect Accounts** button which will take you to the [CloudAEye integrations page][1] (Side Navigation Drawer => Integrations => Datadog).
- Add a Datadog integration. You may have to create your CloudAEye account if you don't have one.

> - Once the keys are populated, your Datadog account will get connected to the CloudAEye.

Reviewer: This will need to be updated with the steps the user needs to take to set up the integration. You will need to remove the steps for OAuth since your integration does not use OAuth. For example, go to the CloudAEye integration in Datadog, click install, etc. This should be the exact flow the user will take to install the integration.

# CloudAEye

## Overview

Kosal from CloudAEye acts as your root cause co-pilot. This integration enables Datadog users to import observability data (logs, metrics, traces) and view automated root cause analyses in CloudAEye. Additionally, it identifies similar issues within Jira and related code changes that may be contributing to the incident. To expedite incident identification, Kosal sends events with a summary and root cause to Datadog.

## Setup

### Installation

- Search for `CloudAEye` in the datadog integrations page.

- Click on **Install Integration** button.

- You need to create a [CloudAEye account](https://docs.cloudaeye.com/user-guide/tasks/register.html) if you do not already have one. 

- On successful registration, login to your cloudaeye account and add a **Datadog Integration**. Follow this [step-by-step guide](https://docs.cloudaeye.com/user-guide/integrations/datadog.html) to setup this integration. 

>   - To integrate your datadog account with CloudAEye, we usually require the following details.

Reviewer 2: done

    - `Site`: The datadog site where your observability data is located. **Ex: US1-East**. Read more about datadog sites [here](https://docs.datadoghq.com/getting_started/site/#access-the-datadog-site)

    - `API Key`: An API key helps us uniquely identify the organization. Read more about API Key [here](https://docs.datadoghq.com/account_management/api-app-keys/)

    - `Application Key`: In addition to an API key, we need an Application Key to programmatically access your data. Read more about Application Key [here](https://docs.datadoghq.com/account_management/api-app-keys/)

> - Once the integration is complete, your Datadog account is successfully connected to CloudAEye.

Reviewer 2: done

### Configuration

> Once integrated, begin exploring your logs, metrics, and traces data in the CloudAEye dashboard. Some of the most useful features are:

Reviewer 2: done

- To get notified about any incident, go to the settings page from CloudAEye's **Side Navigation Drawer => Settings => Root Cause Analysis**, then select the Send Alerts checkbox and choose/add a new notification alert channel of your choice.

> - You can also ask any questions related to your logs, metrics, or traces from Kosal at the [Root Cause Analysis Dashboard][4].

Reviewer 2: done

## Uninstallation

To remove the Datadog integration from CloudAEye:
1. Navigate to the [CloudAEye integrations page][1] (Side Navigation Drawer => Integrations => Datadog) and click **Remove**.
2. Uninstall this integration from Datadog by clicking the **Uninstall Integration** button below. Once you uninstall this integration, any previous authorizations are revoked.

> 3. Ensure that all API keys associated with this integration have been disabled by searching for the integration name (CloudAEye) on the [API Keys management page][3].

Reviewer 3: ```suggestion
3. Ensure that all API keys associated with this integration have been disabled by searching for the integration name (CloudAEye) on the [API keys management page][3].
```

# CloudAEye

## Overview

> Kosal from CloudAEye acts as your root cause co-pilot. This integration enables Datadog users to import observability data (logs, metrics, traces) and view automated root cause analyses. Additionally, it identifies similar issues within Jira and related code changes that may be contributing to the incident. To expedite incident identification, Kosal publishes events with a summary and root cause to a Datadog dashboard.

Reviewer: ```suggestion
Kosal from CloudAEye acts as your root cause co-pilot. This integration enables Datadog users to import observability data (logs, metrics, traces) and view automated root cause analyses in CloudAEye. Additionally, it identifies similar issues within Jira and related code changes that may be contributing to the incident. To expedite incident identification, Kosal sends events with a summary and root cause to Datadog.
```

- Click on **Install Integration** button.

- You need to create a [CloudAEye account](https://docs.cloudaeye.com/user-guide/tasks/register.html) if you do not already have one. 

- On successful registration, login to your cloudaeye account and add a **Datadog Integration**. Follow this [step-by-step guide](https://docs.cloudaeye.com/user-guide/integrations/datadog.html) to setup this integration. 

  - To integrate your datadog account with CloudAEye, we usually require the following details.

    - `Site`: The datadog site where your observability data is located. **Ex: US1-East**. Read more about datadog sites [here](https://docs.datadoghq.com/getting_started/site/#access-the-datadog-site)

    - `API Key`: An API key helps us uniquely identify the organization. Read more about API Key [here](https://docs.datadoghq.com/account_management/api-app-keys/)

    - `Application Key`: In addition to an API key, we need an Application Key to programmatically access your data. Read more about Application Key [here](https://docs.datadoghq.com/account_management/api-app-keys/)

- Once the integration is complete, your Datadog account is successfully connected to CloudAEye.

> 
> 

Reviewer: ```suggestion
```

Reviewer: @ankit-cloudaeye bumpoing

# CloudAEye

## Overview

Kosal from CloudAEye acts as your root cause copilot. This integration enables Datadog users to import observability data (logs, metrics, traces) and view automated root cause analyses in CloudAEye. Additionally, it identifies similar issues within Jira and related code changes that may be contributing to the incident. To expedite incident identification, Kosal sends events with a summary and root cause to Datadog.

## Setup

> ### Pre-installation Steps

Reviewer 3: ```suggestion
### Pre-installation steps
```

- Create a [CloudAEye account](https://docs.cloudaeye.com/user-guide/tasks/register.html) if you do not already have one. 

### Installation

- Log in to your CloudAEye account and add a **Datadog Integration**. 
  - To integrate your Datadog account with CloudAEye, we require the following details.

    - `Site`: The Datadog site where your observability data is located, such as **US1-East**. Read more about Datadog sites in [the documentation](https://docs.datadoghq.com/getting_started/site/#access-the-datadog-site)

    - `API Key`: A Datadog API key helps CloudAEye uniquely identify the organization. Read more about API keys in [the documentation](https://docs.datadoghq.com/account_management/api-app-keys/).

    - `Application Key`: In addition to an API key, CloudAEye needs a Datadog application key to programmatically access your data. Read more about application keys in [the documentation](https://docs.datadoghq.com/account_management/api-app-keys/).

- You can now explore your logs, metrics and traces data in the CloudAEye dashboard.

> ### Notification Setup

Reviewer 3: ```suggestion
### Notification setup
```

To get notified about any incident:
1. On the [CloudAEye integration page][1], navigate to **Settings > Root Cause Analysis** from the side navigation drawer. 
2. Select the **Send Alerts** checkbox.
3. Choose or add the notification alert channel of your choice.

## Uninstallation

To remove the Datadog integration from CloudAEye:
1. From the side navigation drawer on the [CloudAEye integrations page][1], navigate to **Integrations > Datadog**.
2. Click **Remove**.
3. Click the **Uninstall Integration** button. Once you uninstall this integration, any previous authorizations are revoked.

> 3. Ensure that all API keys associated with this integration have been disabled by searching for the integration name (CloudAEye) on the [API keys management page][3].

Reviewer 3: ```suggestion
4. Ensure that all API keys associated with this integration have been disabled by searching for the integration name (CloudAEye) on the [API keys management page][3].
```
